### DISCUSSION TITLE
Stale Development 🙁

### DISCUSSION THREAD

**Discussion by Alfagun74** [caprover/caprover#1544] [General]
I love this project, but the development of it seems kinda stale at this moment.
I would love to see more features and fixes. 
Is there any reason for this or is there anything we all could do to improve this work of art further?

**Comment by drakon** [comment]
Second this question! Just recently discovered CapRover and considering moving quite some apps over from Heroku. 

I really like the general mentality of the project to try to be as light layer over Docker and other tools and giving possibility to configure self when more advanced. 

But I'm a bit hesitant and concerned because of the low activity here which makes me contemplate to still look for other options like Dokku (while currently I'm liking CapRover better). 

So any way to support? For example a donation?

/Edit: With "low activity" I meant in particular that I realised that the last release is over a year old while there have been still quite a couple of commits (I in particular was looking for the updated netdata version; which is quite outdated in the current release).

**Reply by githubsaturn** [reply]
Glad to see this discussion. The project is certainly not unmaintained at all, it's just additional features that have been slowed down a bit lately. The main reason behind that is CapRover goals and scope. As a solo developer of the platform, by design, i do not want to add features that'll be used by less than 5% of users as every new feature is technically adding more work to maintenance. 

CapRover is a heavily stable system and the last thing I want is to add more features to it to break the main premise of it which is reliability. 

At its current scope, I'm personally committed to make sure I address any reliability issues immediately. 

Minor things like netdata update can easily be done by a config override. That's one of the main strengths of caprover (extensibility)

That aside, I'm currently working on an offline branch that adds more features like 2fa etc to the core. Hopefully it'll be released soon.

**Reply by Alfagun74** [reply]
@githubsaturn I see, but there are certainly a lot of UX and UI improvement tickets and general good features open aswell that would be very awesome to have for everyone. It makes me sad nothing really came around for a year.

**Reply by drakon** [reply]
Thanks @githubsaturn for the response! Yes, unmaintained would be an unjust word! 

I do like that there's also not necessarily a lot of changes all the time as I don't like to the underlying platform under my feet changing all the time (and therefore I have to update all the projects depending on it). So I'm totally with you on this! 

Stability should be above everything else, I agree. And Yes, minor things can wait. Didn't know about the override. But I saw it now and will try it. 

I'm also not averse to provide potentially in the future (low critically) features on the frontend (as I'm not too familiar with the whole dev ops/Docker things; that's why I really like CapRover after all!).

**Reply by fu-sen** [reply]
> It makes me sad nothing really came around for a year.

You've seen the [releases](https://github.com/caprover/caprover/releases) and reacted that way. But the [commits](https://github.com/caprover/caprover/commits/master) are brisk. This project is alive and well. You have to look really closely at the project. You should not judge it by Release, not only for this project.

**Comment by maietta** [comment]
If you click on the "watch" button on the projects within CapRover, you'll see that it's certainly not dead.

You have to understand who CapRover was intended for to understand it's not designed to be a traditional hosting solution for your apps. It's a tool to help you deploy apps to a Docker Swarm and offers an ecosystem of community-contributed and maintained one-click apps. 

The best feature of CapRover is that it can be safely removed from your Docker Swarm and your apps continue to run. It's not bloatware and it's not dealing with feature creep. It was not designed to be a user interface for a team of DevOps people. If you are trusted with the password to a Docker node running CapRover, you probably should be considered also trusted with full access to the server because essentially, that's the same thing.

With all that said, I love that CapRover's main focus seems to be squarely aimed at being a solid app container deployment tool. Being what it is, it doesn't need a lot of continuous development to work extremely well. That's not to say that it's dead, either. There's plenty of activity around the project-- even if the core product of the ecosystem isn't as frequently updated.

CapRover and the ecosystem is being actively worked on and contributed to. If people feel that any discussed features should be implemented sooner or other improvements made, this project's maintainer does seem to be open to pull requests. Expanding or  Improving documentation, core project code, one-click apps or other tooling are all open-season for good pull requests.

Like any good project, it starts small and I can't imagine anyone here is getting paid much of anything to build the next big thing. This is one of many hundreds of thousands of projects that exist that run on spare time donated by its contributors.

Remember, if you have the means to contribute money, please do! People have to survive while they build awesome things for others to use for free. It's just how it works in the world of Open Source.

If everyone who used CapRover donated a little something every month, say, half of what they would have paid for a different solution-- I would imagine the project would move a lighting speed.

**Comment by frederikhors** [comment]
@githubsaturn your work is just AMAZING! THANKS!

What worries me a lot is security tough. Even using subdomain, the fact of not having control over those who enter (or try to enter) (even with API) does not make me sleep peacefully.

Can I ask when you're planning to release the news (at least just 2fA)?

The other useful thing would also be to send emails for access authorization.

Thanks again, 🙏

**Reply by githubsaturn** [reply]
Yes. 2FA is coming in coming months!
